sotodlib/site_pipeline/update_obsdb.py

```diff
@@ -0,0 +1,160 @@
+"""update_obsdb
+The config file could be of the form:
+.. code-block:: yaml
+
+    obsdb: dummydb.yaml
+    cols:
+        start_time: float
+        end_time: float
+        n_samples: int
+        telescope: str
+        tube_slot: str
+        type: str
+
+"""
+from sotodlib.core.metadata import ObsDb
+from sotodlib.core import Context 
+from sotodlib.site_pipeline.check_book import main as checkbook
+import os
+import glob
+import yaml
+import numpy as np
+import time
+import argparse
+
+
+def check_meta_type(bookpath):
+    metapath = os.path.join(bookpath, "M_index.yaml")
+    meta = yaml.safe_load(open(metapath, "rb"))
+    if meta is None:
+        return "empty"
+    elif "type" not in meta:
+        return "notype"
+    else:
+        return meta["type"]
+
+def update_obsdb(base_dir, 
+                 config="config.yaml", 
+                 recency=2., 
+                 verbosity=2, 
+                 booktype="both"):
+    """
+        Create or update an obsdb for observation or operations data.
+    Argument
+    ----------
+    base_dir : str
+        The base directory in which to look for books
+    Parameters
+    ----------
+    config : str
+        Path to config file
+    recency : float
+        How far back in time to look for databases, in days. (default: 2.)
+    booktype : str
+        Look for observations or operations data or both (default: both)
+    """
+    bookcart = []
+    bookcartobsdb = ObsDb()
+
+    if booktype=="both":
+        accept_type = ["obs", "oper"]
+    else:
+        accept_type = [booktype]
+
+    configs = yaml.safe_load(open(config, "r"))
+    if "obsdb" in configs:
+        if os.path.isfile(configs["obsdb"]):
+            bookcartobsdb = ObsDb.from_file(configs["obsdb"])
+    if "cols" in configs:
```

Comment:
could we rename "cols" to "obsdb_cols"? This way it's a bit more obvious in a config file where we're defining both obsdb and obsfiledb parameters